openedx/core/process_warnings.py

```diff
@@ -0,0 +1,241 @@
+"""
+Script to process pytest warnings output by pytest-json-report plugin and output it as a html
+"""
+from __future__ import absolute_import
+import json
+import os
+import io
+import pprint
+import re
+import argparse
+from collections import Counter
+import pandas as pd
+from write_to_html import HtmlOutlineWriter  # noqa pylint: disable=import-error,useless-suppression
+from djangolib.markup import HTML, Text  # noqa pylint: disable=import-error
+
+columns = [
+    "message",
+    "category",
+    "filename",
+    "lineno",
+    "high_location",
+    "label",
+    "num",
+    "deprecated",
+]
+columns_index_dict = {key: index for index, key in enumerate(columns)}
+
+
+pp = pprint.PrettyPrinter(indent=4, depth=2)
+
+
+def seperate_warnings_by_location(warnings_data):
+    """
+    Warnings originate from multiple locations, this function takes in list of warning objects
+    and seperates them based on their filename location
+    """
+
+    # first create regex for each know file location
+    warnings_locations = {
+        ".*/python\d\.\d/site-packages/.*\.py": "python",  # noqa pylint: disable=W1401
+        ".*/edx-platform/lms/.*\.py": "lms",  # noqa pylint: disable=W1401
+        ".*/edx-platform/openedx/.*\.py": "openedx",  # noqa pylint: disable=W1401
+        ".*/edx-platform/cms/.*\.py": "cms",  # noqa pylint: disable=W1401
+        ".*/edx-platform/common/.*\.py": "common",  # noqa pylint: disable=W1401
+    }
+
+    # seperate into locations flow:
+    #  * iterate through each wanring_object, see if its filename matches any regex in warning locations.
+    #  * If so, change high_location index on warnings_object to location name
+    for warnings_object in warnings_data:
+        warning_origin_located = False
+        for key in warnings_locations:
+            if (
+                re.search(key, warnings_object[columns_index_dict["filename"]])
```

Comment:
I thought of using those also, but decided I didn't want to add too many new dependencies. But when separated into its own plugin, it would make sense to use something like that.